drivers/applications/drv8825_demo.cpp

```diff
@@ -0,0 +1,37 @@
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+#include <libhal/units.hpp>
+
+#include "../include/drv8825.hpp"
+#include "../hardware_map.hpp"
+
+using namespace hal::literals;
+using namespace std::chrono_literals;
+namespace sjsu::drivers {
+
+void application(application_framework& p_framework)
+{
+  // configure drivers
+  auto& clock = *p_framework.steady_clock;
+  auto& terminal = *p_framework.terminal;
+  
+  auto m_drv8825 = drv8825(
+            *p_framework.out_pin3, 
+            *p_framework.out_pin4, 
+            *p_framework.steady_clock, 
+            drv8825::step_factor::one, 
+            2048, 
+            450us,
+            {p_framework.out_pin0,p_framework.out_pin1,p_framework.out_pin2}
```

Comment:
Might be a good idea to pack this into a struct so its a bit easier to determine which are what. Because this is a pretty long constructor parameter list.